### PR TITLE
Added confidence rating

### PR DESCRIPTION
**Pull request by mmorells**
Added the confidence rating with the logic from the original @greyblake package for R.
Didn't fix tests completely.

### CODE REVIEW

File: detect.go [hunk 1]
```diff
@@ -113,11 +117,49 @@ func detectLangInProfiles(text string, options Options, langProfileList langProf
 	}
 
 	if len(langDistances) == 0 {
-		return -1
+		return -1, 0
 	}
+
 	sort.SliceStable(langDistances, func(i, j int) bool { return langDistances[i].dist < langDistances[j].dist })
+	lang_dist1 := langDistances[0];
```
Comment (abadojack) [inline detect.go]: use camelCase instead when naming variables

File: detect.go [hunk 2]
```diff
@@ -26,17 +26,21 @@ func DetectLangWithOptions(text string, options Options) Lang {
 func DetectWithOptions(text string, options Options) Info {
 	script := DetectScript(text)
 	if script != nil {
-		lang := detectLangBaseOnScript(text, options, script)
+		lang, confidence := detectLangBaseOnScript(text, options, script)
 		return Info{
 			Lang:   lang,
 			Script: script,
+			Confidence: confidence,
```
Comment (abadojack) [inline detect.go]: gofmt your code 

Run `gofmt -w *.go`

File: detect.go [hunk 3]
```diff
@@ -113,11 +117,49 @@ func detectLangInProfiles(text string, options Options, langProfileList langProf
 	}
 
 	if len(langDistances) == 0 {
```
Comment (abadojack) [inline detect.go]: We should probably also check if the length of the slice is 1 to avoid the `index out of range` panic

**Comment by abadojack** [comment]
@mmorells Good work. Could you please make the improvements I suggested

**Comment by mmorells** [comment]
@abadojack thanks! I made the improvements you suggested. I haven't fixed the tests though, didn't have time to look at them. Not exactly sure what's going on in the tests, and I only made the changes (to the tests) needed to compile.

**Comment by greyblake** [comment]
> Added the confidence rating with the logic from the original @greyblake package for R.
> Didn't fix tests completely.

The original package is not for `R`, but `Rust` :)

**Comment by mmorells** [comment]
@greyblake Of course, my bad :D

**Pull request merged**